### DISCUSSION TITLE
How do I ensure a bucket is created?

### DISCUSSION THREAD

**Discussion by Jaysok** [vladimir-bukhtoyarov/bucket4j#457] [Q&A] [answered]
Hi, I'm currently utilizing Bucket4j backed by Redisson. Instead of using Spring Boot, I'm leveraging Spring solely for Dependency Injection, not integrating with Spring-data-redis or similar frameworks for bucket4j integration. I've configured a bucket featuring a Bandwidth parameter set with `capacity=180, refillTokens=20, refillGreedy=true, refillInterval=1s`.

The code snippet below illustrates how I created the bucket:

I create the bucket with the code similar to the following code:
```java
BucketConfiguration bucketConfig = myBucketConfigBuild();
Bucket bucket = proxyManager.builder().build(key, () -> configuration);
createMyInstanceWithBucket(bucket.asBlocking());
```
Upon debugging, I noticed that Bucket4j doesn't seem to generate a Redis key for the bucket until an action is performed on the bucket. As a workaround, I've been using consume(1) immediately after the bucket's creation. This approach serves as a check to ensure the bucket's existence, especially vital when multiple instances attempt to create the same bucket concurrently during deployment.

I'm curious to know:

**Is there a more elegant method to instantiate the actual bucket key in Redis without the need to call consume(1)?**

---

btw, thank you for this amazing lib 🤗

**Comment by vladimir-bukhtoyarov** [comment, marked as answer]
Hello @Jaysok 

Use [getAvailableTokens](https://github.com/bucket4j/bucket4j/blob/master/bucket4j-core/src/main/java/io/github/bucket4j/Bucket.java#L192)

**Reply by Jaysok** [reply]
Thanks!

**Reply by Jaysok** [reply]
Sorry again, the [getAvailableTokens](https://github.com/bucket4j/bucket4j/blob/master/bucket4j-core/src/main/java/io/github/bucket4j/Bucket.java#L192) method is implemented in Bucket but not in BlockingBucket. Could this implementation detail have any impact?

**Reply by vladimir-bukhtoyarov** [reply]
It does not matter. Technically  ```asBlocking``` returns the same bucket.
